eox_core/api/support/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,298 @@
+
+    def test_update_username_in_tenant_not_found(self) -> None:
+        """
+        Test update an edxapp user's username in a tenant that does not exist.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response indicates the user was not found in the tenant.
+        """
+        response = self.update_username(
+            self.tenant_x,
+            {"username": "user-not-found"},
+            {"new_username": "new-username"},
+        )
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response_data["detail"],
+            f"No user found by {{'username': 'user-not-found'}} on site {self.tenant_x['domain']}.",
+        )
+
+    def test_update_username_in_tenant_missing_params(self) -> None:
+        """
+        Test update an edxapp user's username in a tenant without providing the username.
+
+        Expected result:
+        - The status code is 400.
+        - The response indicates the username is required.
+        """
+        response = self.update_username(self.tenant_x)
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_data, ["Email or username needed"])
+
+    def test_update_username_in_tenant_missing_body(self) -> None:
+        """
+        Test update an edxapp user's username in a tenant without providing the new username.
+
+        Expected result:
+        - The status code is 400.
+        - The response indicates the new username is required.
+        """
+        data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, data)
+
+        response = self.update_username(self.tenant_x, params={"username": data["username"]})
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_data, {"new_username": ["This field is required."]})
+
+
+@ddt.ddt
+class TestOauthApplicationAPIIntegration(SupportAPIRequestMixin, BaseIntegrationTest, UsersAPIRequestMixin):
+    """Integration tests for the Oauth Application API."""
+
+    @ddt.data(True, False)
+    def test_create_oauth_application_in_tenant_success(self, create_user: bool) -> None:
+        """
+        Test create an oauth application in a tenant. The user is created if it does not exist.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `create_edxapp_user`
+        - `UserSignupSource`
+
+        Expected result:
+        - The status code is 200.
+        - The response indicates the oauth application was created successfully.
+        """
+        user_data = next(FAKE_USER_DATA)
+        if create_user:
+            self.create_user(self.tenant_x, user_data)
```

Comment:
Could you explain in the docstring why creating the user is needed?

Reply:
The endpoint creates the user if the user doesn't exist. The conditional is to validate both cases. I updated the docstring: https://github.com/eduNEXT/eox-core/commit/bb37aab1015654d52b843dfae5ecc542dc8f9576